mlar/tests/integration.rs

```diff
@@ -129,6 +129,104 @@ fn ensure_directory_content(directory: &Path, files: &[NamedTempFile]) {
     assert_eq!(fname2content.len(), 0);
 }
 
+#[test]
+fn test_create_from_dir() {
+    let mlar_file = NamedTempFile::new("output.mla").unwrap();
+    let ecc_public = Path::new("../samples/test_x25519_pub.pem");
+    let ecc_private = Path::new("../samples/test_x25519.pem");
+
+    // Temporary directory to test recursive file addition
+    let tmp_dir = TempDir::new().unwrap();
+    let mut subfile1_path = tmp_dir.path().join("subfile1");
+    let subdir_path = tmp_dir.path().join("subdir");
+    let mut subfile2_path = subdir_path.join("subfile2");
+
+    std::fs::write(&mut subfile1_path, "Test1").unwrap();
+    std::fs::create_dir(subdir_path).unwrap();
+    std::fs::write(&mut subfile2_path, "Test2").unwrap();
+
+    // `mlar create -o output.mla -p samples/test_x25519_pub.pem <tmp_dir>`
+    let mut cmd = Command::cargo_bin(UTIL).unwrap();
+    cmd.arg("create")
+        .arg("-o")
+        .arg(mlar_file.path())
+        .arg("-p")
+        .arg(ecc_public);
+
+    cmd.arg(tmp_dir.path());
+
+    let mut file_list = String::new();
+    // The exact order of the files in the archive depends on the order of the
+    // result of `read_dir` which is plateform and filesystem dependent.
+    for entry in read_dir(tmp_dir.path()).unwrap() {
+        let path = entry.unwrap().path();
+        if path.is_dir() {
+            file_list.push_str(format!("{}\n", subfile2_path.to_string_lossy()).as_str());
```

Comment:
IMHO, it would be more understandable and prone to changes if this part was actually parsing and retrieving filenames from the path.
In this current version, the code is actually only checking for the presence of one file and one directory.

Maybe a version looking like the one you already done in `mlar` (ie. recursive listing of files) would be better, and would allow one to add / modify the layout of the test directory structure.

Reply:
I agree. I've changed it accordingly in the tests commit.